code/modules/cm_marines/dropship_ammo.dm

```diff
@@ -348,4 +348,43 @@
 		spawn(5)
 			for(var/turf/T in range(2, impact))
 				if(!locate(/obj/flamer_fire) in T) // No stacking flames!
-					new/obj/flamer_fire(T)
\ No newline at end of file
+					new/obj/flamer_fire(T)
+
+/obj/structure/ship_ammo/minirocket/illumination
+	name = "illumination rocket-launched flare stack"
+	desc = "A pack of laser guided mini rockets, each loaded with a payload of white-star illuminant and a parachute, while extremely ineffective at damaging the enemy, it is very effective at lighting the battlefield so marines can damage the enemy."
+	icon_state = "minirocket_ilm"
+	point_cost = 250
+
+	detonate_on(turf/impact)
+		impact.ceiling_debris_check(2)
+		spawn(5)
+			var/turf/T = pick(range(5, impact))
+			explosion(T,-1,-1,1, 2, 0)// Smaller explosion to prevent this becoming the PO meta
+			var/datum/effect_system/expl_particles/P = new/datum/effect_system/expl_particles()
+			P.set_up(4, 0, T)
+			P.start()
+			spawn(5)
+				var/datum/effect_system/smoke_spread/S = new/datum/effect_system/smoke_spread()
+				S.set_up(1,0,T,null)
+				S.start()
+			new/obj/item/device/flashlight/flare/on/cas(T) // thx
+			if(!ammo_count && loc)
+				cdel(src) //deleted after last minirocket is fired and impact the ground.
+
+/obj/item/device/flashlight/flare/on/cas
+	name = "illumination flare"
+	desc = "Report this if you actually see this FUCK"
+	icon_state = "" //No sprite
+	invisibility = 101
+	mouse_opacity = 0
+	brightness_on = 8 //Magnesium/sodium fires (White star) really are bright
+
+	New()
+		..()
+		var/turf/T = get_turf(src)
+		fuel = rand(1000, 1100) // About the same burn time as a flare, considering it requires it's own CAS run.
+		T.visible_message("<span class='warning'>You see a flare go up in the sky!</span>")
+		playsound(T, 'sound/weapons/gun_flare.ogg', 50, 1, 4) // stolen from the mortar i'm not even sorry
+		spawn(10) // Sloppy i know but the only way to fix this.
+			update_brightness()
```

Comment:
I remember you saying earlier you found out the smoke is what was causing the lighting issues. Why are you then having the lighting update happen outside the thing causing the need for an update?

Reply:
I found out that having the lighting update again afterwards is a good way to fix many bugs with it. Not just the smoke bug.

Reply:
Outside the scope of this PR, the issues are with CM's particle system and how it treats turfs when it comes to explosions.

Reply:
It's not outside the scope to fix issues that are causing your feature to not work properly

Reply:
But my feature has been tested to work properly when using the delays.

Reply:
Doesn't matter, working code != good code